ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiLayeredWorldState.java

```diff
@@ -129,15 +124,21 @@ public UInt256 getStorageValue(final Address address, final UInt256 key) {
     while (currentLayer != null) {
       final Optional<UInt256> maybeValue =
           currentLayer.trieLog.getStorageBySlotHash(address, slotHash);
-      if (maybeValue.isPresent()) {
+      final Optional<UInt256> maybeOriginalValue =
+          currentLayer.trieLog.getOriginalStorageBySlotHash(address, slotHash);
```

Comment:
I don't think this is a good place for getOriginalStorageBySlotHash.  Original in the sense of EIP-2200 gas costing.  I think we need to go to renaming BonsaiValue `getOriginal` as `getPrior` to remove that confusion.

Reply:
indeed I wanted to rename it in another PR but why not do it now